### PR TITLE
Add Githab PR request template

### PR DESCRIPTION
**Pull request by Rulasmur**
# Description
This PR adds a PR Template for use with github

# Testing
Create a PR in the github templates branch this PR is coming from

# Outstanding Work
- [ ] Leaving this in here to show the template
- [ ] But normally this would be removed unless there is outstanding work

# Notes
> Leaving this in to show the pull structure of the PR Template
> The template actually includes comments as well

### Pre Pull Request Checklist:

- [x] Code has been scanned with SonarLint and has passed([Find it here](http://www.sonarlint.org/intellij/))
- [x] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
- [x] There are no errors present in the project
- [x] Code has been formatted and indented
- [x] Methods have appropriate Javadoc([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!-- Thanks for submitting a pull request for Destination Sol! :-)
+Please fill in some brief details below about the PR.
+If it contains source code please make sure to do everything in the pre pull request checklist first.
+If you add unit tests we'll love you forever! -->
```
Comment (rzats) [inline .github/PULL_REQUEST_TEMPLATE.md]: DS doesn't have a unit testing framework - is this line still OK to keep?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# Outstanding Work
+- [ ] If there is work still outstanding, please include it here
+- [ ] if there is nothing outstanding, remove this section.
+
+# Notes
+> Include any extra notes here
+
+### Pre Pull Request Checklist:
+<!-- When scanning the code with SonarLint, we don't except you to fix every single thing, just fix yours, and a few extra-->
```
Comment (rzats) [inline .github/PULL_REQUEST_TEMPLATE.md]: "just don't introduce any new issues, and maybe even fix a few existing ones" sounds better IMO.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Code has been scanned with SonarLint and has passed([Find it here](http://www.sonarlint.org/intellij/))
```
Comment (rzats) [inline .github/PULL_REQUEST_TEMPLATE.md]: Missing a space here after "passed". Also, hyperlinking the word "SonarLint" to the project page looks slightly better.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
+- [ ] There are no errors present in the project
+- [ ] Code has been formatted and indented
```
Comment (rzats) [inline .github/PULL_REQUEST_TEMPLATE.md]: Aren't both of the above covered by SonarLint?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Methods have appropriate Javadoc([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))
```
Comment (rzats) [inline .github/PULL_REQUEST_TEMPLATE.md]: Missing another space here.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Rulasmur) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe it'll inspire someone to add it?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (Rulasmur) [inline .github/PULL_REQUEST_TEMPLATE.md]: Nope, SonarLint just covers the quality analysis stuff

**Pull request merged**